app/src/main/java/io/github/gsantner/memetastic/activity/MainActivity.java

```diff
@@ -220,6 +228,25 @@ protected void onCreate(Bundle savedInstanceState) {
         }
     }
 
+    public void toggleHiddenTab() {
+        MenuItem hiddenItem = _navigationView.getMenu().findItem(R.id.action_mode_hidden);
+
+        List<MemeData.Image> imageList = new ArrayList<>();
+
+        for (String hidden : app.settings.getHiddenMemesTemplate()) {
+            MemeData.Image image = MemeData.findImage(new File(hidden));
+            if (image != null) {
+                imageList.add(image);
+            }
```

Comment:
we can break at this point. all we need is the info if at least one hidden item is in there

Reply:
Even when there's no hidden meme, the array returned from settings isn't empty it contains a single string - white space and when there's one item it's still an array of one string, that's why i.had to resort to a list of images.

Another choice would be to check it the first string's image is null

Reply:
i meant breaking insiide the condition, not outside. `findimage != null` says the file/meme does exist / is valid

Reply:
Okay, i understand now. Would refactor it